tests/test_logging.py::test_rich_print_forcecolor fails if NO_COLOR=1 is set in the calling environment

The test suite fails if `NO_COLOR=1` is set in the environment (to prevent pytest from printing colors):

```pytb
$ NO_COLOR=1 nox -e tests
nox > Running session tests
nox > Creating virtual environment (virtualenv) using python3.12 in .nox/tests
nox > python -m pip install '-e.[test,test-meta,test-numpy]'
nox > pytest 
========================================================= test session starts =========================================================
platform linux -- Python 3.12.0b3, pytest-7.3.2, pluggy-1.2.0
installed packages of interest: build==0.10.0 cattrs==23.1.2 hatch-fancy-pypi-readme==23.1.0 packaging==23.1 pathspec==0.11.1 pip==23.1.2 pybind11==2.10.4 pyproject-metadata==0.7.1 pytest==7.3.2 pytest-subprocess==1.5.0 scikit_build_core==0.4.7.dev2+gd928992.d20230623 setuptools==68.0.0 setuptools-scm==7.1.0 typing-extensions==4.6.3 wheel==0.40.0
sysconfig platform: linux-x86_64
rootdir: /tmp/scikit-build-core
configfile: pyproject.toml
testpaths: tests
plugins: subprocess-1.5.0
collected 131 items / 1 skipped                                                                                                       

tests/test_builder.py .........xs.........                                                                                      [ 15%]
tests/test_cmake_config.py ....                                                                                                 [ 18%]
tests/test_custom_modules.py .                                                                                                  [ 19%]
tests/test_dynamic_metadata.py ........                                                                                         [ 25%]
tests/test_file_processor.py .                                                                                                  [ 25%]
tests/test_fileapi.py ....                                                                                                      [ 29%]
tests/test_generator_default.py ...                                                                                             [ 31%]
tests/test_get_requires.py .......                                                                                              [ 36%]
tests/test_logging.py .F                                                                                                        [ 38%]
tests/test_module_dir.py ..                                                                                                     [ 39%]
tests/test_name_main.py ...                                                                                                     [ 41%]
tests/test_prepare_metadata.py ..                                                                                               [ 43%]
tests/test_process_scripts.py .                                                                                                 [ 44%]
tests/test_program_search.py ss...                                                                                              [ 48%]
tests/test_pyproject_abi3.py .                                                                                                  [ 48%]
tests/test_pyproject_extra_dirs.py ..                                                                                           [ 50%]
tests/test_pyproject_pep517.py ........s.                                                                                       [ 58%]
tests/test_pyproject_pep518.py ......                                                                                           [ 62%]
tests/test_pyproject_pep660.py ...                                                                                              [ 64%]
tests/test_settings.py .......................                                                                                  [ 82%]
tests/test_setuptools_abi3.py .                                                                                                 [ 83%]
tests/test_setuptools_pep517.py ...                                                                                             [ 85%]
tests/test_setuptools_pep518.py ..                                                                                              [ 87%]
tests/test_shutil.py ...                                                                                                        [ 89%]
tests/test_simple_pure.py ...                                                                                                   [ 91%]
tests/test_simplest_c.py ...                                                                                                    [ 93%]
tests/test_skbuild_settings.py ......                                                                                           [ 98%]
tests/test_wheelfile_utils.py ..                                                                                                [100%]

============================================================== FAILURES ===============================================================
_____________________________________________________ test_rich_print_forcecolor ______________________________________________________

capsys = <_pytest.capture.CaptureFixture object at 0x7f6820fe6000>
monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x7f6820fe7f50>

    def test_rich_print_forcecolor(capsys, monkeypatch):
        monkeypatch.setenv("FORCE_COLOR", "1")
        rich_print("[red bold]hello[/bold] world", end="")
>       assert capsys.readouterr().out == "\33[91m\33[1mhello\33[22m world\33[0m"
E       AssertionError: assert 'hello world' == '\x1b[91m\x1b... world\x1b[0m'
E         - hello world
E         + hello world

tests/test_logging.py:13: AssertionError
======================================================= short test summary info =======================================================
SKIPPED [1] tests/test_fortran.py:17: could not import 'numpy': No module named 'numpy'
SKIPPED [1] tests/test_builder.py:66: MSVC only
SKIPPED [1] tests/test_program_search.py:17: could not import 'cmake': No module named 'cmake'
SKIPPED [1] tests/test_program_search.py:26: could not import 'ninja': No module named 'ninja'
SKIPPED [1] tests/test_pyproject_pep517.py:242: Doesn't work yet
XFAIL tests/test_builder.py::test_get_python_library - Doesn't matter if this fails, usually not used
FAILED tests/test_logging.py::test_rich_print_forcecolor - AssertionError: assert 'hello world' == '\x1b[91m\x1b... world\x1b[0m'
=================================== 1 failed, 125 passed, 5 skipped, 1 xfailed in 267.57s (0:04:27) ===================================
nox > Command pytest  failed with exit code 1
nox > Session tests failed.
```

--- a/tests/test_logging.py
+++ b/tests/test_logging.py
@@ -3,7 +3,7 @@ from scikit_build_core._logging import rich_print
 
 def test_rich_print_nocolor(capsys, monkeypatch):
     monkeypatch.setenv("NO_COLOR", "1")
-    monkeypatch.delenv("FORCE_COLOR")
+    monkeypatch.delenv("FORCE_COLOR", raising=False)
     rich_print("[red]hello[/red] world", end="")
     assert capsys.readouterr().out == "hello world"
 
@@ -17,6 +17,6 @@ def test_rich_print_nocolor_forcecolor(capsys, monkeypatch):
 
 def test_rich_print_forcecolor(capsys, monkeypatch):
     monkeypatch.setenv("FORCE_COLOR", "1")
-    monkeypatch.delenv("NO_COLOR")
+    monkeypatch.delenv("NO_COLOR", raising=False)
     rich_print("[red bold]hello[/bold] world", end="")
     assert capsys.readouterr().out == "\33[91m\33[1mhello\33[22m world\33[0m"

--- a/tests/test_logging.py
+++ b/tests/test_logging.py
@@ -3,11 +3,20 @@ from scikit_build_core._logging import rich_print
 
 def test_rich_print_nocolor(capsys, monkeypatch):
     monkeypatch.setenv("NO_COLOR", "1")
+    monkeypatch.delenv("FORCE_COLOR")
+    rich_print("[red]hello[/red] world", end="")
+    assert capsys.readouterr().out == "hello world"
+
+
+def test_rich_print_nocolor_forcecolor(capsys, monkeypatch):
+    monkeypatch.setenv("NO_COLOR", "1")
+    monkeypatch.setenv("FORCE_COLOR", "1")
     rich_print("[red]hello[/red] world", end="")
     assert capsys.readouterr().out == "hello world"
 
 
 def test_rich_print_forcecolor(capsys, monkeypatch):
     monkeypatch.setenv("FORCE_COLOR", "1")
+    monkeypatch.delenv("NO_COLOR")
     rich_print("[red bold]hello[/bold] world", end="")
     assert capsys.readouterr().out == "\33[91m\33[1mhello\33[22m world\33[0m"

